### PR TITLE
test(countDocuments): full test coverage for countDocuments

### PR DESCRIPTION
**Pull request by rweinberger**
Add tests that were node-specific or missing from spec tests.

Fixes NODE-1559

### CODE REVIEW

**Comment by rweinberger** [comment]
Let me know if there are other cases to test that I missed!

File: test/functional/collection_tests.js [hunk 1]
```diff
@@ -1616,6 +1620,141 @@ describe('Collection', function() {
...
+    it('countDocuments should return appropriate error if aggregation fails', function(done) {
+      const replyHandler = () => {};
+      const cb = err => {
+        expect(err).to.exist;
+        expect(err.errmsg).to.equal('aggregation error');
+      };
+
+      testSkipLimit(
+        {
+          replyHandler: replyHandler,
```
Comment (daprahamian) [inline test/functional/collection_tests.js]: You can use the shorthand object notation for these:

```js
testSkipLimit({
  replyHandler,
  cb,
  options: {},
  ...
```

File: test/functional/collection_tests.js [hunk 2]
```diff
@@ -1616,6 +1620,141 @@ describe('Collection', function() {
...
+
+    beforeEach(() => {
+      return mock.createServer().then(s => {
+        server = s;
+      });
+    });
+
+    afterEach(() => mock.cleanup());
+
+    function testSkipLimit(config, done) {
```
Comment (daprahamian) [inline test/functional/collection_tests.js]: should this still be called `testSkipLimit` if it is being used in `'countDocuments should return appropriate error if aggregation fails'`?

Comment (rweinberger) [reply test/functional/collection_tests.js]: good catch, my bad

File: lib/operations/collection_ops.js [hunk 3]
```diff
@@ -230,12 +230,9 @@ function countDocuments(coll, query, options, callback) {
...
   delete options.skip;
 
-  coll.aggregate(pipeline, options, (err, result) => {
+  coll.aggregate(pipeline, options).toArray((err, docs) => {
     if (err) return handleCallback(callback, err);
-    result.toArray((err, docs) => {
-      if (err) return handleCallback(err);
-      handleCallback(callback, null, docs.length ? docs[0].n : 0);
-    });
+    handleCallback(callback, null, docs.length ? docs[0].n : 0);
```
Comment (EdikDolynskyi) [inline lib/operations/collection_ops.js]: If `docs` is an empty array there will be an error

Comment (daprahamian) [reply lib/operations/collection_ops.js]: @EdikDolynskyi I don't think that is the case here. if docs is an empty array, `docs.length` will be 0.

**Pull request merged**